clippy_lints/src/bool_assert_comparison.rs

```diff
@@ -90,13 +103,28 @@ impl<'tcx> LateLintPass<'tcx> for BoolAssertComparison {
                         }
 
                         let non_eq_mac = &mac[..mac.len() - 3];
+                        let expr_string = if lit_value ^ is_eq {
+                            format!("!({})", source::snippet(cx, other_expr.span, ""))
+                        } else {
+                            source::snippet(cx, other_expr.span, "").to_string()
+                        };
+
+                        let suggestion = if args.len() <= 2 {
+                            format!("{}!({})", non_eq_mac, expr_string)
+                        } else {
+                            let mut str = format!("{}!({}", non_eq_mac, expr_string);
+                            for el in &args[2..] {
+                                str = format!("{}, {}", str, source::snippet(cx, el.span, ""));
+                            }
+                            format!("{})", str)
```

Comment:
Use the `String` API (`push` and `push_str`).

Or better yet, get just one snippet of all the format args with something like

```rust
match fmt_args {
    [] => None,
    [a] => Some(a.span),
    [a, .., b] => Some(a.span.to(b.span)),
}